symforce/codegen/codegen_util.py

```diff
@@ -342,8 +342,14 @@ def get_formatted_list(
             flattened_value = [value]
         elif issubclass(arg_cls, geo.Matrix):
             if isinstance(config, codegen_config.PythonConfig):
-                # TODO(nathan): Not sure this works for 2D matrices
-                formatted_symbols = [sm.Symbol(f"{key}[{j}]") for j in range(storage_dim)]
+                # NOTE(hayk): Python codegen currently seems broken for matrix inputs, because we
+                # pass a 2D numpy array then index it linearly, which breaks since it returns a
+                # whole column. This modification fixes that, but is broken for vector inputs that
+                # are either a sequence or a 1D numpy array. Discuss and fix.
+                formatted_symbols = []
+                for j in range(value.shape[1]):
+                    for i in range(value.shape[0]):
+                        formatted_symbols.append(sm.Symbol(f"{key}[{i}, {j}]"))
```

Comment:
@aaron-skydio @bradley-solliday-skydio I think there's a bug here that we should discuss and work out

Reply:
I may have addressed it here: https://github.com/symforce-org/symforce/pull/159/commits/dc803873f27b5256159fe30777b27be1a418a3f6

I'm not sure if this breaks anything with Python codegen yet. There's a decision in play here whether 1D geo vectors get generated as lists, 1D numpy arrays, or 2D numpy arrays.

Reply:
Yeah cc @nathan-skydio too, I don't think we want to generate them as lists but could see either 1d or 2d numpy arrays

Reply:
Maybe generating 2D arrays makes the most sense given that a `geo.Matrix` always has the number of rows/cols defined, so I'm not sure a 1D `geo.Matrix` really exists (i.e. I think `geo.V3.shape` is `(3,1)`, so we might want `my_generated_numpy_vec3.shape` to be `(3,1)` instead of `(3,)`). I don't feel super strongly one way or the other though

Reply:
With dc80387 we can do this easily per language though right?  I think it makes sense for python to use all-2d numpy arrays, since that's easy, but for JS to use 1D arrays for column vectors

Reply:
Yes we can do 2D numpy arrays for python and use 1D arrays in JS. Just slightly odd to have different apis like that

Reply:
Personally I lean towards just making things 2d everywhere if we want uniformity

Reply:
Yeah I'm a bit torn on the uniformity vs idiomatic use. I wonder if we might just want to introduce a dependency like https://github.com/nicolaspanel/numjs